npm/translatePolicy.go

```diff
@@ -273,10 +278,21 @@ func translateIngress(ns string, targetSelector metav1.LabelSelector, rules []ne
 		}
 
 		// fromRuleExists
-		for _, fromRule := range rule.From {
+		for j, fromRule := range rule.From {
 			// Handle IPBlock field of NetworkPolicyPeer
 			if fromRule.IPBlock != nil {
 				if len(fromRule.IPBlock.CIDR) > 0 {
+					ipCidrs[i] = append(ipCidrs[i], fromRule.IPBlock.CIDR)
+					cidrIpsetName := name + "-cidr" + strconv.Itoa(i) + "in"
+					if len(fromRule.IPBlock.Except) > 0 {
+						for _, except := range fromRule.IPBlock.Except {
+							ipCidrs[i] = append(ipCidrs[i], except + " " + util.IpsetNomatch)
+						}
+						addedIngressFromEntry = true
+					}
+					if j != 0 {
```

Comment:
Besides IPBLocks, we also have  podselector and namespace selector in the following steps, so we cannot continue to next rule item.

Reply:
We may later refactor this lengthy function to be more clean and easy to maintain.
CC @jaer-tsun

Reply:
> Besides IPBLocks, we also have podselector and namespace selector in the following steps, so we cannot continue to next rule item.

We can continue. Please note the continue condition is when j != 0 which means the podselector and namespace selector has already been added in iptable entry when j = 0. It's a key point to improve our performance. When j = 0 which means npm loop through the 1st item in ipBlock, npm will create a cidr ipset, create a iptable entry. When j > 0, npm just need to add cidr to ipset entry, no need to create iptable entries anymore.

Reply:
> We may later refactor this lengthy function to be more clean and easy to maintain.
> CC @jaer-tsun

Agree, I drafted a PR to refactor translatePolicy.go to remove duplicated code between translateIngress and translateEgress and make it concise.